reference.Spec.Hostname() always panic if no "/" presents

**Description**

`reference.Spec.Hostname()` always panic if no `/` presents in the Locator.

**Steps to reproduce the issue:**
```golang
package main

import (
	"fmt"
	"log"

	"github.com/containerd/containerd/reference"
)

func main() {
	ref := "test:1234"
	spec, err := reference.Parse(ref)
	if err != nil {
		log.Fatalf("fail to parse %s: %v", ref, err)
	}
	hostname := spec.Hostname()
	fmt.Println(hostname)
}
```
Code available at [the Go playground](https://play.golang.org/p/8bFOGrdTmp5)

**Describe the results you received:**
```
panic: runtime error: slice bounds out of range [:10] with length 9

goroutine 1 [running]:
github.com/containerd/containerd/reference.Spec.Hostname(0x4986d0, 0x9, 0x0, 0x0, 0x0, 0x0)
	/tmp/gopath583841917/pkg/mod/github.com/containerd/containerd@v1.3.0/reference/reference.go:129 +0x80
main.main()
	/tmp/sandbox478338582/prog.go:18 +0x80
```

**Describe the results you expected:**
It should print `test`.

**Output of `containerd --version`:**
Current `master` branch or tag `1.3.0`.

--- a/reference/reference.go
+++ b/reference/reference.go
@@ -124,7 +124,7 @@ func (r Spec) Hostname() string {
 	i := strings.Index(r.Locator, "/")
 
 	if i < 0 {
-		i = len(r.Locator) + 1
+		return r.Locator
 	}
 	return r.Locator[:i]
 }

